activesupport/lib/active_support/testing/method_call_assertions.rb

```diff
@@ -13,11 +13,18 @@ def assert_called(object, method_name, message = nil, times: 1)
           assert_equal times, times_called, error
         end
 
-        def assert_called_with(object, method_name, args = [], returns: nil)
+        def assert_called_with(object, method_name, args = [], returns: nil, use_distinct_returns: false)
           mock = Minitest::Mock.new
 
           if args.all? { |arg| arg.is_a?(Array) }
-            args.each { |arg| mock.expect(:call, returns, arg) }
+            if use_distinct_returns
+              if returns.nil? || !returns.is_a?(Array) || returns.length != args.length
```

Comment:
not sure.. we can probably compare nil to type array directly?

if so we can remove returns.nil?